src/game/server/gamecontext.cpp

```diff
@@ -2915,6 +2920,52 @@ void CGameContext::ConClearVotes(IConsole::IResult *pResult, void *pUserData)
 	}
 }
 
+struct CMapNameItem
+{
+	char m_aName[MAX_PATH_LENGTH - 4];
+
+	bool operator<(const CMapNameItem &Other) const { return str_comp_nocase(m_aName, Other.m_aName) < 0; }
+};
+
+void CGameContext::ConAddMapVotes(IConsole::IResult *pResult, void *pUserData)
+{
+	CGameContext *pSelf = (CGameContext *)pUserData;
+
+	sorted_array<CMapNameItem> MapList;
+	pSelf->Storage()->ListDirectory(IStorage::TYPE_ALL, "maps", MapScan, &MapList);
+
+	for(int i = 0; i < MapList.size(); i++)
+	{
+		char aDescription[64];
+		str_format(aDescription, sizeof(aDescription), "Map: %s", MapList[i].m_aName);
+
+		char aCommand[MAX_PATH_LENGTH * 2 + 10];
+		char aMapEscaped[MAX_PATH_LENGTH * 2];
+		char *pDst = aMapEscaped;
+		str_escape(&pDst, MapList[i].m_aName, aMapEscaped + sizeof(aMapEscaped));
+		str_format(aCommand, sizeof(aCommand), "change_map \"%s\"", aMapEscaped);
+
+		pSelf->AddVote(aDescription, aCommand);
+	}
+
+	pSelf->Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "server", "added maps to votes");
+}
+
+int CGameContext::MapScan(const char *pName, int IsDir, int DirType, void *pUserData)
+{
+	sorted_array<CMapNameItem> *pMapList = (sorted_array<CMapNameItem> *)pUserData;
+
+	if(IsDir || !str_endswith(pName, ".map"))
+		return 0;
+
+	CMapNameItem Item;
+	int Length = str_length(pName);
+	str_copy(Item.m_aName, pName, minimum(static_cast<int>(sizeof(Item.m_aName)), Length - 3));
```

Comment:
```suggestion
	str_truncate(Item.m_aName, sizeof(Item.m_aName), pName, Length - 4);
```

Makes the intent clearer IMO.

Reply:
Done.